File: ch01/exercises/exercise-1.6.1.3/main.ml
let sumk2 n =
  let open Float in
  List.init n (fun k -> k + 1)
  |> List.map (fun k -> 1. /. pow (of_int k) 2.)
  |> List.fold_left ( +. ) 0.

let () =
  let open Float in
  let open Printf in
  let series_sum = pow pi 2. /. 6. in
  let approx_ans = sumk2 1_000 in
  printf "%f\n%f\n" series_sum approx_ans

File: ch01/exercises/exercise-1.6.1.3/Main.hs
module Main where

main :: IO ()
main = do
  print seriesSum
  print approxAns

sumk2 :: Integer -> Double
sumk2 n = sum $ map (\k -> 1 / (fromIntegral k ^ 2)) [1 .. n - 1]

seriesSum = pi ^ 2 / 6

approxAns = sumk2 100_000
